### ISSUE TITLE
Clarify docs for pod affinity and anti-affinity

### ISSUE THREAD

**Issue by bassam** [rook/rook#1121]
The Rook operator should manage the count of MONs in the cluster and automatically set the anti-affinity rules as needed. There could a few discrete steps:

- Less that 3 nodes in the cluster -- the Operator should just run 1 MON with no anti-affinity
- 3 nodes or more -- the Operator should create 3 MONs by default and set anti-affinity. It should also manage adding more MONs as the cluster grows.

Ceph guidelines are somewhat vague here:

```
For an initial deployment of a multi-node Ceph cluster, Red Hat requires three Monitors, increasing the number two at a time if a valid need for more than three Monitors exists.
```

Also the anti-affinity rules should take into account zone and region information if available spreading the MONs across failure domains.

**Comment by travisn** [comment]
The operator does explicit placement of mons on nodes. I'm not sure why the ant-affinity to other mons would be interesting since the operator is already placing the mons on different nodes, as long as there are enough nodes.

**Comment by bassam** [comment]
@travisn why do we document anti-affinity rules in our yaml then?

**Comment by travisn** [comment]
the anti-affinity could still be used for avoiding placement of mons on the same nodes as other types of pods that the admin is aware of. It just doesn't need to be set for the mons themselves.

**Comment by bassam** [comment]
@travisn ok. so by default the operator will do the right thing for MONs, OSDs etc. If there are other interactions the admin can set their own affinity. Sound right? Would be good to document.

**Comment by DanKerns** [comment]
Sounds like an opportunity to improve the documentation...

**Comment by travisn** [comment]
Sure, we can add more description to https://rook.io/docs/rook/master/cluster-crd.html#placement-configuration-settings

**Comment by galexrt** [comment]
**monCount**:
If not specified by the user, I would like the idea to start one mon per node in a cluster with 3 or less nodes. The default if not specified with 3 or more nodes should be `3`.

**Affinity, Placement**:
I don't see that there is any commonly used "account zone and region information" which is reliable across platforms.
@bassam comment to find "labels" for this information together with the Kubernetes community, is a good idea and implement it later.

The current implementation of `podAntiAffinity` requires the user to add it to the Cluster CRD, which currently lacks an example in the documentation (as pointed out by @DanKerns and @travisn).
I don't think that the `podAntiAffinity` should be set automatically by the operator or at least be a toggleable feature then (when implemented).
I personally use the `podAntiAffinity` to as @travisn wrote to prevent the placement of two mons on the same node, in case a node for mons fails.

I personally think that such things as `podAntiAffinity` (and `*affinity`) should at least then be able to be overwritten by the user, if we are going to implement changes to the logic.

**Comment by travisn** [comment]
Will be fixed by #1372
